330 - Implement IDS Contextual Action Panel

EdwardCoyle opened the pull request:
**Explain the details for making this change. What existing problem does the pull request solve?**
This PR adds the IDS Contextual Action Panel component
- Extends Modal component
- Implements a toolbar area
- The Toolbar's contents are connected to the `onButtonClick` callback provided by the Modal
- Makes adjustments to Popup component that fix placement coordinates to account for components nested inside positioned parent elements.
- Adds docs/types

**Related github/jira issue (required)**:
Closes #330

**Steps necessary to review your pull request (required)**:
- Pull, build, run
- Open http://localhost:4300/ids-contextual-action-panel and click the trigger to open
- See that all form controls and buttons inside the CAP work (close/save are wired to hide the CAP)

**Included in this Pull Request**:
- [x] An e2e or functional test for the bug or feature.
- [x] A note to the change log.

tmcconechy commented on demos/ids-contextual-action-panel/example.html:
```diff
@@ -0,0 +1,101 @@
+      </ids-button>
+    </ids-toolbar-section>
+
+    <!--
```
Any reason to keep?

tmcconechy commented on src/components/ids-contextual-action-panel/ids-contextual-action-panel.js:
```diff
@@ -0,0 +1,89 @@
+} from '../../core/ids-element';
+
+import IdsModal from '../ids-modal';
+import '../ids-toolbar';
```
Should it import something? import idsToolbar from  '../ids-toolbar';

EdwardCoyle replied:
I'm just depending on it here for the template.  I'm not using the named export (not invoking with `new`) so didn't use it.

tmcconechy commented on src/components/ids-contextual-action-panel/ids-contextual-action-panel.js:
```diff
@@ -0,0 +1,89 @@
+ * IDS Contextual Action Panel
+ * @type {IdsContextualActionPanel}
+ * @inherits IdsModal
+ * @mixes IdsEventsMixin
```
Doesnt directly mix so not sure if needed?

tmcconechy reviewed (commented):
Looks awesome 👍🏻  
- Only a couple simple comments
- In terms of functionality one thing that stands out is when you click the CAP buttons the "ripple effect" is much darker before. Maybe needs a lighter style like https://main-enterprise.demo.design.infor.com/components/contextualactionpanel/example-index.html
- i also noticed on about/cap ect. that clicking a button to close has a short delay before actually closing - might be an issue in the future?

EdwardCoyle commented:
@tmcconechy I addressed all the comments here except for this one:

> i also noticed on about/cap ect. that clicking a button to close has a short delay before actually closing - might be an issue in the future?

tmcconechy commented:
ok, maybe we might end up shortening that delay a bit as it might "slow down" the workflow. But we can wait for further feedback. I wasn't sure on it

EdwardCoyle commented:
@tmcconechy Just rebased this against main, shouldn't have any changes

jmacaluso711 reviewed (approved):
Looks good! 👍
